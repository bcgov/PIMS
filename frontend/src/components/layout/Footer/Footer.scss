.footer {
  bottom: 0;
  padding: 10px 0;

  a {
    display: inline;
    &:link,
    &:visited {
      color: #ffffff;
      font-size: 16px;
      font-family: 'BCSans', Fallback, sans-serif;
      padding: 0 10px;
      border-right: 1px solid #666;
    }
    &:hover,
    &:focus {
      text-decoration: underline;
    }
  }
}
