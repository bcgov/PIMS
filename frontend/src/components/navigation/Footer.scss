.footer {
  background-color: #036;
  color: #fff;
  bottom: 0;
  width: 100%;
  padding: 10px 15px;
  border-top: 2px solid #fcba19;
  .content {
    margin: 0 80% 0 20%;
  }

  a:link{
    color: #ffffff;
    font-size: 16px;
    font-family: "BCSans" Fallback, sans-serif;
    padding: 0 10px;
    border-right: 1px solid #666;
  }

  a:visited{
    color: #ffffff;
    font-size: 16px;
    font-family: "BCSans" Fallback, sans-serif;
    padding: 0 10px;
    border-right: 1px solid #666;
  }
}
