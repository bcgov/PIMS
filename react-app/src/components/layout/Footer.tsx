import { footerCopyright } from '@/constants/strings';
import { Box, Link, Toolbar, Typography, useTheme } from '@mui/material';
import React from 'react';

const Footer = (): JSX.Element => {
  const theme = useTheme();
  return (
    <footer>
      <Toolbar
        sx={{ height: '74px', borderTop: '1px solid', borderTopColor: theme.palette.gray.main }}
      >
        <Typography>{footerCopyright}</Typography>
        <Box flexGrow={1} />
        <Box display={'flex'} gap={'32px'}>
          <Link href="#" variant="h5">
            Accessibility
          </Link>
          <Link href="#" variant="h5">
            Privacy
          </Link>
          <Link href="#" variant="h5">
            Copyright
          </Link>
          <Link href="#" variant="h5">
            Disclaimer
          </Link>
        </Box>
      </Toolbar>
    </footer>
  );
};

export default Footer;
