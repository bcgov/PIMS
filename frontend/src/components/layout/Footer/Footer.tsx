import './Footer.scss';

import React from 'react';
import { Nav, Navbar } from 'react-bootstrap';

function Footer() {
  return (
    <Navbar expand className="footer">
      <Nav.Link href="http://www.gov.bc.ca/gov/content/home/disclaimer">Disclaimer</Nav.Link>
      <Nav.Link href="http://www.gov.bc.ca/gov/content/home/privacy">Privacy</Nav.Link>
      <Nav.Link href="http://www.gov.bc.ca/gov/content/home/accessible-government">
        Accessibility
      </Nav.Link>
      <Nav.Link href="http://www.gov.bc.ca/gov/content/home/copyright">Copyright</Nav.Link>
    </Navbar>
  );
}

export default Footer;
