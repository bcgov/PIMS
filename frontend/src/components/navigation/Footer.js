import './Footer.scss';
import React from 'react';

function Footer(props) {

  return (
    <footer className="footer">
    </footer>
  );
}

export default Footer;


