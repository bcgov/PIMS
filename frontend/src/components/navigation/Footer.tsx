import './Footer.scss';
import React from 'react';

function Footer() {

  return (
    <footer className="footer">
      <div className="content">
        <a href="http://www.gov.bc.ca/gov/content/home/disclaimer">Disclaimer</a>
        <a href="http://www.gov.bc.ca/gov/content/home/privacy">Privacy</a>
        <a href="http://www.gov.bc.ca/gov/content/home/accessible-government">Accessibility</a>
        <a href="http://www.gov.bc.ca/gov/content/home/copyright">Copyright</a>
      </div>
    </footer>
  );
}

export default Footer;


