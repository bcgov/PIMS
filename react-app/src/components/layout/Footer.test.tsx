import { create } from 'react-test-renderer';
import React from 'react';
import Footer from '@/components/layout/Footer';

jest.mock('@mui/material', () => ({
  ...jest.requireActual('@mui/material'),
  useTheme: () => ({
    palette: {
      gray: {
        main: '#FFFFFF',
      },
    },
  }),
}));

describe('Footer.tsx', () => {
  it('should match the existing snapshot', () => {
    const tree = create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
