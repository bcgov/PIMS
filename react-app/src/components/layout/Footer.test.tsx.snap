// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer.tsx should match the existing snapshot 1`] = `
<footer>
  <div
    className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-ydkvb7-MuiToolbar-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Â© 2024 Government of British Columbia.
    </p>
    <div
      className="MuiBox-root css-i9gxme"
    />
    <div
      className="MuiBox-root css-13t3iqc"
    >
      <a
        className="MuiTypography-root MuiTypography-h5 MuiLink-root MuiLink-underlineAlways css-1qas1pa-MuiTypography-root-MuiLink-root"
        href="#"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        Accessibility
      </a>
      <a
        className="MuiTypography-root MuiTypography-h5 MuiLink-root MuiLink-underlineAlways css-1qas1pa-MuiTypography-root-MuiLink-root"
        href="#"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        Privacy
      </a>
      <a
        className="MuiTypography-root MuiTypography-h5 MuiLink-root MuiLink-underlineAlways css-1qas1pa-MuiTypography-root-MuiLink-root"
        href="#"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        Copyright
      </a>
      <a
        className="MuiTypography-root MuiTypography-h5 MuiLink-root MuiLink-underlineAlways css-1qas1pa-MuiTypography-root-MuiLink-root"
        href="#"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        Disclaimer
      </a>
    </div>
  </div>
</footer>
`;
